src/python/pants/reporting/metrics.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import logging
+from typing import Dict, List
+
+from pants.option.subsystem import Subsystem
+
+logger = logging.getLogger(__name__)
+
+
+class MetricsReporting(Subsystem):
+    """Subsystem to aggregate and display internal Pants metrics."""
+
+    options_scope = "pants-metrics"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._counters: Dict[str, int] = {}
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--report",
+            type=bool,
+            default=False,
+            help="Whether internal Pants metrics should be reported at the end of a run.",
+        )
+
+    def handle_workunits(
+        self,
+        *,
+        completed_workunits: List[Dict],
+        started_workunits: List[Dict],
+        context,
```

Comment:
Can you add type annotations on these two values?

Reply:
Done.